Mathlib/Topology/Algebra/Module/StrongTopology.lean

```diff
@@ -72,125 +72,151 @@ variable {𝕜₁ 𝕜₂ : Type*} [NormedField 𝕜₁] [NormedField 𝕜₂] (
-#align continuous_linear_map.strong_topology.embedding_coe_fn ContinuousLinearMap.strongTopology.embedding_coeFn
-
-theorem strongUniformity.uniformAddGroup [UniformSpace F] [UniformAddGroup F] (𝔖 : Set (Set E)) :
-    @UniformAddGroup (E →SL[σ] F) (strongUniformity σ F 𝔖) _ := by
-  letI : UniformSpace (E →SL[σ] F) := strongUniformity σ F 𝔖
-  let φ : (E →SL[σ] F) →+ E →ᵤ[𝔖] F :=
-    ⟨⟨(DFunLike.coe : (E →SL[σ] F) → E →ᵤ[𝔖] F), rfl⟩, fun _ _ => rfl⟩
-  exact (strongUniformity.uniformEmbedding_coeFn _ _ _).uniformAddGroup φ
-#align continuous_linear_map.strong_uniformity.uniform_add_group ContinuousLinearMap.strongUniformity.uniformAddGroup
-
-theorem strongTopology.topologicalAddGroup [TopologicalSpace F] [TopologicalAddGroup F]
-    (𝔖 : Set (Set E)) : @TopologicalAddGroup (E →SL[σ] F) (strongTopology σ F 𝔖) _ := by
+#align continuous_linear_map.strong_uniformity.uniform_embedding_coe_fn UniformConvergenceCLM.uniformEmbedding_coeFn
+
+theorem embedding_coeFn [UniformSpace F] [UniformAddGroup F] (𝔖 : Set (Set E)) :
+    Embedding (X := UniformConvergenceCLM σ F 𝔖) (Y := E →ᵤ[𝔖] F)
+      (UniformOnFun.ofFun 𝔖 ∘ DFunLike.coe) :=
+  UniformEmbedding.embedding (uniformEmbedding_coeFn _ _ _)
+#align continuous_linear_map.strong_topology.embedding_coe_fn UniformConvergenceCLM.embedding_coeFn
+
+instance instAddCommGroup [TopologicalSpace F] [TopologicalAddGroup F] (𝔖 : Set (Set E)) :
+    AddCommGroup (UniformConvergenceCLM σ F 𝔖) := ContinuousLinearMap.addCommGroup
+
+instance instUniformAddGroup [UniformSpace F] [UniformAddGroup F] (𝔖 : Set (Set E)) :
+    UniformAddGroup (UniformConvergenceCLM σ F 𝔖) := by
+  let φ : (UniformConvergenceCLM σ F 𝔖) →+ E →ᵤ[𝔖] F :=
+    ⟨⟨(DFunLike.coe : (UniformConvergenceCLM σ F 𝔖) → E →ᵤ[𝔖] F), rfl⟩, fun _ _ => rfl⟩
+  exact (uniformEmbedding_coeFn _ _ _).uniformAddGroup φ
+#align continuous_linear_map.strong_uniformity.uniform_add_group UniformConvergenceCLM.instUniformAddGroup
+
+instance instTopologicalAddGroup [TopologicalSpace F] [TopologicalAddGroup F]
+    (𝔖 : Set (Set E)) : TopologicalAddGroup (UniformConvergenceCLM σ F 𝔖) := by
   letI : UniformSpace F := TopologicalAddGroup.toUniformSpace F
   haveI : UniformAddGroup F := comm_topologicalAddGroup_is_uniform
-  letI : UniformSpace (E →SL[σ] F) := strongUniformity σ F 𝔖
-  haveI : UniformAddGroup (E →SL[σ] F) := strongUniformity.uniformAddGroup σ F 𝔖
   infer_instance
-#align continuous_linear_map.strong_topology.topological_add_group ContinuousLinearMap.strongTopology.topologicalAddGroup
+#align continuous_linear_map.strong_topology.topological_add_group UniformConvergenceCLM.instTopologicalAddGroup
 
-theorem strongTopology.t2Space [TopologicalSpace F] [TopologicalAddGroup F] [T2Space F]
-    (𝔖 : Set (Set E)) (h𝔖 : ⋃₀ 𝔖 = Set.univ) : @T2Space (E →SL[σ] F) (strongTopology σ F 𝔖) := by
+theorem t2Space [TopologicalSpace F] [TopologicalAddGroup F] [T2Space F]
```

Comment:
I think this is okay now, no?
```suggestion
instance t2Space [TopologicalSpace F] [TopologicalAddGroup F] [T2Space F]
```

Reply:
I think that doesn't ever fire, because in every concrete case we still have to show `⋃₀ 𝔖 = Set.univ` and together with that proof we want to have an instance.